lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -53,6 +53,24 @@ class LoadBalancer
 end
 end
 
+require 'chef/provider/load_balancer'
+class Chef
+class Provider
+  class LoadBalancer
+    # We override this so we can specify a machine name as `i-123456`
+    # This is totally a hack until we move away from base resources
+    def get_machine_spec!(machine_name)
```

Comment:
I would prefer that users don't have to add flags to start getting a bug fix.  I also don't think it will interfere with other providers - they would also have to use IDs that match the same regex.